File: GridView/VirtualRadGridView/VirtualRadGridViewVB/VirtualRadGridViewSortingFiltering.Core/GridViewCellValueEventArgsEx.vb

Imports System.Collections.Generic
Imports System.Linq
Imports System.Reflection
Imports System.Text
Imports Telerik.WinControls.UI

Namespace VirtualRadGridViewSortingFiltering.Core
    Public Class GridViewCellValueEventArgsEx
        Inherits GridViewCellValueEventArgs
        Private Shared oldArgsType As Type
        Private Shared rowInfoField As FieldInfo

        Private m_rowInfo As GridViewRowInfo

        Public Property OldArgs() As GridViewCellValueEventArgs
            Get
                Return m_OldArgs
            End Get
            Private Set(value As GridViewCellValueEventArgs)
                m_OldArgs = Value
            End Set
        End Property
        Private m_OldArgs As GridViewCellValueEventArgs

        Public ReadOnly Property RowInfo() As GridViewRowInfo
            Get
                If Me.m_rowInfo Is Nothing Then
                    Me.m_rowInfo = DirectCast(rowInfoField.GetValue(Me.OldArgs), GridViewRowInfo)
                End If

                Return Me.m_rowInfo
            End Get
        End Property

        Shared Sub New()
            oldArgsType = GetType(GridViewCellValueEventArgs)
            rowInfoField = oldArgsType.GetField("rowInfo", BindingFlags.Instance Or BindingFlags.NonPublic)
        End Sub

        Public Sub New(oldArgs As GridViewCellValueEventArgs, columnIndex As Integer, rowIndex As Integer)
            MyBase.New(columnIndex, rowIndex)
            Me.OldArgs = oldArgs
        End Sub
    End Class
End Namespace

'=======================================================
'Service provided by Telerik (www.telerik.com)
'Conversion powered by NRefactory.
'Twitter: @telerik
'Facebook: facebook.com/telerik
'=======================================================

File: GridView/VirtualRadGridView/VirtualRadGridViewSortingFilteringAfterQ3_2015/VirtualRadGridViewSortingFilteringAfterQ3_2015CS/VirtualRadGridViewSortingFiltering.Core/GridViewCellValueEventArgsEx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using System.Text;
using Telerik.WinControls.UI;

namespace VirtualRadGridViewSortingFiltering.Core
{
    public class GridViewCellValueEventArgsEx : GridViewCellValueEventArgs
    {
        private static Type oldArgsType;
        private static FieldInfo rowInfoField;

        private GridViewRowInfo rowInfo;

        public GridViewCellValueEventArgs OldArgs { get; private set; }

        public GridViewRowInfo RowInfo
        {
            get
            {
                if (this.rowInfo == null)
                {
                    this.rowInfo = (GridViewRowInfo)rowInfoField.GetValue(this.OldArgs);
                }

                return this.rowInfo;
            }
        }

        static GridViewCellValueEventArgsEx()
        {
            oldArgsType = typeof(GridViewCellValueEventArgs);
            rowInfoField = oldArgsType.GetField("rowInfo", BindingFlags.Instance | BindingFlags.NonPublic);
        }

        public GridViewCellValueEventArgsEx(GridViewCellValueEventArgs oldArgs, int columnIndex, int rowIndex)
            : base(columnIndex, rowIndex)
        {
            this.OldArgs = oldArgs;
        }
    }
}
